### DISCUSSION TITLE
希望可以添加ts视频文件的在线观看

### DISCUSSION THREAD

**Discussion by falcon-hayaki** [spencerwooo/onedrive-vercel-index#660] [General]
不知道web端在线观看实现起来难度怎么样，如果不方便的话底下也可以加一个potplayer打开的链接。现在打开会一直转，底下也只有下载链接

**Comment by myl7** [comment]
最新版的 ts 文件是可以被正常识别为视频并且有“用 PotPlayer 打开”的按钮的。
Web 端在线观看 ts 视频太麻烦了，基本不用考虑，即便有解决方案也会引入很多部件。

**Reply by falcon-hayaki** [reply]
诶 但我打开我盘里的ts文件会一直加载，而且没有potplayer打开的按钮..是文件有问题吗
![image](https://user-images.githubusercontent.com/39900178/170460218-0f4503b2-f958-4eda-9d54-483ba89a164f.png)

**Reply by myl7** [reply]
那应该是你的 ts 文件被识别为 TypeScript 代码文件了。
本项目区分 ts 视频和 ts 代码的方法是看 OneDrive 识别出的文件类型。
出现这种状况可能是因为 OneDrive 没有把这份 ts 文件识别为视频。

如果你希望所有 ts 文件都被识别为视频的话，可以自行修改 [`getPreviewType`](https://github.com/spencerwooo/onedrive-vercel-index/blob/main/utils/getPreviewType.ts) 的判断逻辑。
